code/modules/food_and_drinks/kitchen_machinery/griddle.dm

```diff
@@ -90,8 +121,21 @@
 	update_appearance()
 	update_grill_audio()
 
+/obj/machinery/griddle/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
+	UnregisterSignal(get_turf(old_loc), COMSIG_ATOM_HITBY)
+	RegisterSignal(get_turf(src), COMSIG_ATOM_HITBY, .proc/AddThrownItemToGrill)
+	. = ..()
```

Comment:
These need to be null checked

Reply:
if(.) or the the signals on the turfs existing?